src/buildblock/inverse_SSRB.cxx

```diff
@@ -31,85 +33,102 @@ inverse_SSRB(ProjData& proj_data_4D,
-			{		
-				sino_4D = proj_data_4D.get_empty_sinogram(out_ax_pos_num, out_segment_num);				
-				const float out_m = 
-					proj_data_4D_info_sptr->
-					get_m(Bin(out_segment_num, 0, out_ax_pos_num, 0));				
-				int num_contributing_sinos = 0;
-								
-				for (int in_ax_pos_num = proj_data_3D.get_min_axial_pos_num(0); 
-				     in_ax_pos_num  <= proj_data_3D.get_max_axial_pos_num(0);
-				     ++in_ax_pos_num )
-				{
-					sino_3D = proj_data_3D.get_sinogram(in_ax_pos_num,0);					
-					const float in_m = 
-						proj_data_3D_info_sptr->get_m(Bin(0, 0, in_ax_pos_num, 0));
-
-                    
-					if (fabs(out_m - in_m) < 1E-2)
-					{
-					        ++num_contributing_sinos;
-					        sino_4D += sino_3D;
-						if (proj_data_4D.set_sinogram(sino_4D) == Succeeded::no)
-							return Succeeded::no;
-						break;
-					}
-					const float in_m_next = in_ax_pos_num == proj_data_3D.get_max_axial_pos_num(0) ? 
-						-1000000.F : proj_data_3D_info_sptr->get_m(Bin(0, 0, in_ax_pos_num+1, 0));
-
-					if (fabs(out_m - .5F*(in_m + in_m_next)) < 1E-2)
-					{
-					        ++num_contributing_sinos;
-						sino_4D += sino_3D;
-						sino_4D += proj_data_3D.get_sinogram(in_ax_pos_num+1,0);
-						sino_4D *= .5F;
-						if (proj_data_4D.set_sinogram(sino_4D) == Succeeded::no)
-							return Succeeded::no;
-						break;
-					}
-				}
-				if (num_contributing_sinos == 0)
-				  warning("inverse_SSRB: no sinogram contributes to segment %d, axial_pos_num %d",
-					  out_segment_num, out_ax_pos_num);
+  {
+    for (int out_ax_pos_num = proj_data_4D.get_min_axial_pos_num(out_segment_num); 
+        out_ax_pos_num <= proj_data_4D.get_max_axial_pos_num(out_segment_num);
+        ++out_ax_pos_num )
+    {		
+      sino_4D = proj_data_4D.get_empty_sinogram(out_ax_pos_num, out_segment_num);				
+      const float out_m = proj_data_4D_info_sptr->
+                            get_m(Bin(out_segment_num, 0, out_ax_pos_num, 0));
+              
+      // Go through all direct sinograms to check which pair are closest.
+      bool sinogram_set = false;
+      for (int in_ax_pos_num = proj_data_3D.get_min_axial_pos_num(0); 
+           in_ax_pos_num <= proj_data_3D.get_max_axial_pos_num(0);
+           ++in_ax_pos_num)
+      {
+        // for the first slice there is no previous
+        const auto distance_to_previous = in_ax_pos_num == proj_data_3D.get_min_axial_pos_num(0) ? 
+                                          std::numeric_limits<float>::max() : abs(out_m - in_m.at(in_ax_pos_num - 1));
+        const auto distance_to_current = abs(out_m - in_m.at(in_ax_pos_num));
+        // for the last slice there is no next
+        const auto distance_to_next = in_ax_pos_num == proj_data_3D.get_max_axial_pos_num(0) ? 
+                                      std::numeric_limits<float>::max() : abs(out_m - in_m.at(in_ax_pos_num + 1));
+        if (distance_to_current <= distance_to_previous && 
+            distance_to_current <= distance_to_next)
```

Comment:
Not convinced that this condition is correct, but it's too late in the day to think...